src/main/java/jaffa/mco364/paint/PencilTool.java

```diff
@@ -12,28 +13,30 @@ public PencilTool(PaintProperties properties) {
 	}
 
 	@Override
-	public void mousePressed(Graphics g, int x, int y) {
+	public void mousePressed(Graphics2D g, int x, int y) {
 		g.setColor(properties.getColor());
+		g.setStroke(new BasicStroke(properties.getWeight()));
```

Comment:
dont create a new BasicStroke object every time the mouse is pressed. Move BasicStroke into PaintProperties and change also on line 32.